packages/serverpod/test/database/extensions/comparison_warning_test.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:serverpod/src/database/migrations/table_comparison_warning.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('ComparisonWarning toString output', () {
+    test(
+      'Given a column with no subs when toString is called then the correct output is returned',
+      () {
+        var warning = ColumnComparisonWarning(
+          mismatch: 'missing column',
+          expected: 'age',
+          found: 'none',
+        );
+
+        expect(
+          warning.toString(),
+          equals(
+            '''Column missing column mismatch:
+ - expected "age", found "none".''',
+          ),
+        );
+      },
+    );
+
+    test(
+      'Given a column with one sub when toString is called then the correct output is returned',
+      () {
+        var subWarning = ColumnComparisonWarning(
+          mismatch: 'type',
+          expected: 'integer',
+          found: 'text',
+        );
+
+        var warning = ColumnComparisonWarning().addSub(subWarning);
+
+        expect(
+          warning.toString(),
+          equals(
+            '''Column mismatch:
+   - type mismatch:
+     - expected "integer", found "text".''',
+          ),
+        );
+      },
+    );
+
+    test(
+      'Given a column with multiple subs when toString is called then the correct output is returned',
+      () {
+        var subWarning1 = ColumnComparisonWarning(
+          mismatch: 'type',
+          expected: 'integer',
+          found: 'text',
+        );
+
+        var subWarning2 = ColumnComparisonWarning(
+          mismatch: 'nullability',
+          expected: 'false',
+          found: 'true',
+        );
+
+        var warning =
+            ColumnComparisonWarning().addSubs([subWarning1, subWarning2]);
+
+        expect(
+          warning.toString(),
+          equals(
+            '''Column mismatch:
+   - type mismatch:
+     - expected "integer", found "text".
```

Comment:
Question/Discussion: Would this message be more clear if it was simply `- Column type mismatch: expected "integer", found "text."`?

We are redundantly repeating mismatch a bunch of times now.